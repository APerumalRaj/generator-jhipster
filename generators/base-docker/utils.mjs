import os from 'os';
import { exec } from 'child_process';
import { buildToolTypes } from '../../jdl/index.js';

const { GRADLE } = buildToolTypes;
const isWin32 = os.platform() === 'win32';

/**
 * build a generated application.
 *
 * @param {String} buildTool - maven | gradle
 * @param {String} profile - dev | prod
 * @param {Boolean} buildWar - build a war instead of a jar
 * @param {Function} cb - callback when build is complete
 * @returns {object} the command line and its result
 */
export const buildApplication = (buildTool, profile, buildWar, cb) => {
  let buildCmd = 'mvnw -ntp verify -B';

  if (buildTool === GRADLE) {
    buildCmd = 'gradlew';
    if (buildWar) {
      buildCmd += ' bootWar';
    } else {
      buildCmd += ' bootJar';
    }
  }
  if (buildWar) {
    buildCmd += ' -Pwar';
  }

  if (!isWin32) {
    buildCmd = `./${buildCmd}`;
  }
  buildCmd += ` -P${profile}`;
  return {
    stdout: exec(buildCmd, { maxBuffer: 1024 * 10000 }, cb).stdout,
    buildCmd,
  };
};
