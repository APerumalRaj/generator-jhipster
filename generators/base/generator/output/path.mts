import normalize from 'normalize-path';

export function normalizeOutputPath(outputPath: any): any {
  return outputPath ? normalize(outputPath) : outputPath;
}

export function applyPathCustomizer(context: any, outputPath: any, outputPathCustomizer: any): any {
  if (Array.isArray(outputPathCustomizer)) {
    let outputhPathApplied = outputPath;
    outputPathCustomizer.forEach(customizer => {
      outputhPathApplied = customizer.call(context, outputhPathApplied);
    });
    return outputhPathApplied;
  }
  return outputPathCustomizer.call(context, outputPath);
}
