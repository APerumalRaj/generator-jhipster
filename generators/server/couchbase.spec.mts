import { jestExpect as expect } from 'mocha-expect-snapshot';
import lodash from 'lodash';
import { basename, dirname, join } from 'path';
import { fileURLToPath } from 'url';

import { testBlueprintSupport, serverSamples, extendMatrix, entitiesSimple as entities } from '../../test/support/index.mjs';
import Generator from './index.js';
import { skipPrettierHelpers as helpers } from '../../test/utils/utils.mjs';

import DatabaseTypes from '../../jdl/jhipster/database-types';

const { snakeCase } = lodash;

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const generator = basename(__dirname);
const generatorFile = join(__dirname, 'index.js');

const { COUCHBASE: databaseType } = DatabaseTypes;
const commonConfig = { databaseType, baseName: 'jhipster', nativeLanguage: 'en', languages: ['fr', 'en'] };

const couchbaseSamples = extendMatrix(serverSamples, {
  searchEngine: ['no', 'couchbase'],
});

const testSamples = (): [string, any][] =>
  Object.entries(couchbaseSamples).map(([name, sample]) => [
    name,
    {
      defaults: true,
      applicationWithEntities: {
        config: {
          ...sample,
          ...commonConfig,
        },
        entities,
      },
    },
  ]);

describe(`JHipster ${databaseType} generator`, () => {
  it('generator-list constant matches folder name', async () => {
    await expect((await import('../generator-list.js')).default[`GENERATOR_${snakeCase(generator).toUpperCase()}`]).toBe(generator);
  });
  it('should support features parameter', () => {
    const instance = new Generator([], { help: true }, { bar: true });
    expect(instance.features.bar).toBe(true);
  });
  describe('blueprint support', () => testBlueprintSupport(generator));

  testSamples().forEach(([name, sample]) => {
    const sampleConfig = sample.applicationWithEntities.config;
    const { authenticationType } = sampleConfig;

    describe(name, () => {
      let runResult;

      before(async () => {
        runResult = await helpers.create(generatorFile).withOptions(sample).run();
      });

      after(() => runResult.cleanup());

      it('should match generated files snapshot', () => {
        expect(runResult.getStateSnapshot()).toMatchSnapshot();
      });
      it('contains correct authenticationType', () => {
        runResult.assertFileContent('.yo-rc.json', new RegExp(`"authenticationType": "${authenticationType}"`));
      });
      it('contains correct databaseType', () => {
        runResult.assertFileContent('.yo-rc.json', new RegExp(`"databaseType": "${databaseType}"`));
      });
    });
  });
});
