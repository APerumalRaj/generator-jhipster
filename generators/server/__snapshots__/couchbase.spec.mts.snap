// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JHipster couchbase generator samples matrix should match snapshot 1`] = `
Object {
  "gateway-jwt-gradle-enableTranslation(true)-com.package-fix--Rest-skipCommitHook(true)-websocket(false)-skipUserManagement(false)-serviceDiscoveryType(no)-couchbase": Object {
    "applicationWithEntities": Object {
      "config": Object {
        "applicationType": "gateway",
        "authenticationType": "jwt",
        "baseName": "jhipster",
        "buildTool": "gradle",
        "databaseType": "couchbase",
        "dtoSuffix": "Rest",
        "enableTranslation": true,
        "entitySuffix": "",
        "jhiPrefix": "fix",
        "languages": Array [
          "fr",
          "en",
        ],
        "nativeLanguage": "en",
        "packageName": "com.package",
        "searchEngine": "couchbase",
        "serverSideOptions": Array [],
        "serviceDiscoveryType": "no",
        "skipCommitHook": true,
        "skipUserManagement": false,
        "testFrameworks": Array [],
        "websocket": false,
      },
      "entities": Array [
        Object {
          "changelogDate": "20220129000100",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "Simple",
        },
        Object {
          "changelogDate": "20220129000200",
          "clientRootFolder": "test-root",
          "dto": "mapstruct",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "AnotherSimple",
          "pagination": "pagination",
          "service": "serviceImpl",
        },
      ],
    },
    "defaults": true,
  },
  "gateway-oauth2-maven-enableTranslation(false)-tech.jhipster-jhi-Entity-DTO-skipCommitHook(false)-cucumber-enableSwaggerCodegen:true-websocket(true)-consul-searchEngine(no)": Object {
    "applicationWithEntities": Object {
      "config": Object {
        "applicationType": "gateway",
        "authenticationType": "oauth2",
        "baseName": "jhipster",
        "buildTool": "maven",
        "databaseType": "couchbase",
        "dtoSuffix": "DTO",
        "enableTranslation": false,
        "entitySuffix": "Entity",
        "jhiPrefix": "jhi",
        "languages": Array [
          "fr",
          "en",
        ],
        "nativeLanguage": "en",
        "packageName": "tech.jhipster",
        "searchEngine": "no",
        "serverSideOptions": Array [
          "enableSwaggerCodegen:true",
        ],
        "serviceDiscoveryType": "consul",
        "skipCommitHook": false,
        "testFrameworks": Array [
          "cucumber",
        ],
        "websocket": true,
      },
      "entities": Array [
        Object {
          "changelogDate": "20220129000100",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "Simple",
        },
        Object {
          "changelogDate": "20220129000200",
          "clientRootFolder": "test-root",
          "dto": "mapstruct",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "AnotherSimple",
          "pagination": "pagination",
          "service": "serviceImpl",
        },
      ],
    },
    "defaults": true,
  },
  "microservice-jwt-reactive(false)-maven-enableTranslation(false)-tech.jhipster-jhi-Entity-DTO-skipCommitHook(false)-cucumber-websocket(false)-skipUserManagement(false)-serviceDiscoveryType(no)-searchEngine(no)": Object {
    "applicationWithEntities": Object {
      "config": Object {
        "applicationType": "microservice",
        "authenticationType": "jwt",
        "baseName": "jhipster",
        "buildTool": "maven",
        "databaseType": "couchbase",
        "dtoSuffix": "DTO",
        "enableTranslation": false,
        "entitySuffix": "Entity",
        "jhiPrefix": "jhi",
        "languages": Array [
          "fr",
          "en",
        ],
        "nativeLanguage": "en",
        "packageName": "tech.jhipster",
        "reactive": false,
        "searchEngine": "no",
        "serverSideOptions": Array [],
        "serviceDiscoveryType": "no",
        "skipCommitHook": false,
        "skipUserManagement": false,
        "testFrameworks": Array [
          "cucumber",
        ],
        "websocket": false,
      },
      "entities": Array [
        Object {
          "changelogDate": "20220129000100",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "Simple",
        },
        Object {
          "changelogDate": "20220129000200",
          "clientRootFolder": "test-root",
          "dto": "mapstruct",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "AnotherSimple",
          "pagination": "pagination",
          "service": "serviceImpl",
        },
      ],
    },
    "defaults": true,
  },
  "microservice-jwt-reactive(true)-gradle-enableTranslation(true)-com.package-fix--Rest-skipCommitHook(true)-enableSwaggerCodegen:true-skipUserManagement(true)-eureka-couchbase": Object {
    "applicationWithEntities": Object {
      "config": Object {
        "applicationType": "microservice",
        "authenticationType": "jwt",
        "baseName": "jhipster",
        "buildTool": "gradle",
        "databaseType": "couchbase",
        "dtoSuffix": "Rest",
        "enableTranslation": true,
        "entitySuffix": "",
        "jhiPrefix": "fix",
        "languages": Array [
          "fr",
          "en",
        ],
        "nativeLanguage": "en",
        "packageName": "com.package",
        "reactive": true,
        "searchEngine": "couchbase",
        "serverSideOptions": Array [
          "enableSwaggerCodegen:true",
        ],
        "serviceDiscoveryType": "eureka",
        "skipCommitHook": true,
        "skipUserManagement": true,
        "testFrameworks": Array [],
      },
      "entities": Array [
        Object {
          "changelogDate": "20220129000100",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "Simple",
        },
        Object {
          "changelogDate": "20220129000200",
          "clientRootFolder": "test-root",
          "dto": "mapstruct",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "AnotherSimple",
          "pagination": "pagination",
          "service": "serviceImpl",
        },
      ],
    },
    "defaults": true,
  },
  "microservice-oauth2-reactive(false)-maven-enableTranslation(false)-tech.jhipster-jhi-Entity-DTO-skipCommitHook(false)-gatling-enableSwaggerCodegen:true-websocket(true)-eureka-searchEngine(no)": Object {
    "applicationWithEntities": Object {
      "config": Object {
        "applicationType": "microservice",
        "authenticationType": "oauth2",
        "baseName": "jhipster",
        "buildTool": "maven",
        "databaseType": "couchbase",
        "dtoSuffix": "DTO",
        "enableTranslation": false,
        "entitySuffix": "Entity",
        "jhiPrefix": "jhi",
        "languages": Array [
          "fr",
          "en",
        ],
        "nativeLanguage": "en",
        "packageName": "tech.jhipster",
        "reactive": false,
        "searchEngine": "no",
        "serverSideOptions": Array [
          "enableSwaggerCodegen:true",
        ],
        "serviceDiscoveryType": "eureka",
        "skipCommitHook": false,
        "testFrameworks": Array [
          "gatling",
        ],
        "websocket": true,
      },
      "entities": Array [
        Object {
          "changelogDate": "20220129000100",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "Simple",
        },
        Object {
          "changelogDate": "20220129000200",
          "clientRootFolder": "test-root",
          "dto": "mapstruct",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "AnotherSimple",
          "pagination": "pagination",
          "service": "serviceImpl",
        },
      ],
    },
    "defaults": true,
  },
  "microservice-oauth2-reactive(true)-gradle-enableTranslation(true)-com.package-fix--Rest-skipCommitHook(true)-gatling-consul-couchbase": Object {
    "applicationWithEntities": Object {
      "config": Object {
        "applicationType": "microservice",
        "authenticationType": "oauth2",
        "baseName": "jhipster",
        "buildTool": "gradle",
        "databaseType": "couchbase",
        "dtoSuffix": "Rest",
        "enableTranslation": true,
        "entitySuffix": "",
        "jhiPrefix": "fix",
        "languages": Array [
          "fr",
          "en",
        ],
        "nativeLanguage": "en",
        "packageName": "com.package",
        "reactive": true,
        "searchEngine": "couchbase",
        "serverSideOptions": Array [],
        "serviceDiscoveryType": "consul",
        "skipCommitHook": true,
        "testFrameworks": Array [
          "gatling",
        ],
      },
      "entities": Array [
        Object {
          "changelogDate": "20220129000100",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "Simple",
        },
        Object {
          "changelogDate": "20220129000200",
          "clientRootFolder": "test-root",
          "dto": "mapstruct",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "AnotherSimple",
          "pagination": "pagination",
          "service": "serviceImpl",
        },
      ],
    },
    "defaults": true,
  },
  "monolith-jwt-reactive(false)-maven-enableTranslation(false)-tech.jhipster-jhi-Entity-DTO-skipCommitHook(false)-enableSwaggerCodegen:true-websocket(true)-skipUserManagement(false)-consul-searchEngine(no)": Object {
    "applicationWithEntities": Object {
      "config": Object {
        "applicationType": "monolith",
        "authenticationType": "jwt",
        "baseName": "jhipster",
        "buildTool": "maven",
        "databaseType": "couchbase",
        "dtoSuffix": "DTO",
        "enableTranslation": false,
        "entitySuffix": "Entity",
        "jhiPrefix": "jhi",
        "languages": Array [
          "fr",
          "en",
        ],
        "nativeLanguage": "en",
        "packageName": "tech.jhipster",
        "reactive": false,
        "searchEngine": "no",
        "serverSideOptions": Array [
          "enableSwaggerCodegen:true",
        ],
        "serviceDiscoveryType": "consul",
        "skipCommitHook": false,
        "skipUserManagement": false,
        "testFrameworks": Array [],
        "websocket": true,
      },
      "entities": Array [
        Object {
          "changelogDate": "20220129000100",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "Simple",
        },
        Object {
          "changelogDate": "20220129000200",
          "clientRootFolder": "test-root",
          "dto": "mapstruct",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "AnotherSimple",
          "pagination": "pagination",
          "service": "serviceImpl",
        },
      ],
    },
    "defaults": true,
  },
  "monolith-jwt-reactive(true)-gradle-enableTranslation(true)-com.package-fix--Rest-skipCommitHook(true)-cucumber-skipUserManagement(true)-serviceDiscoveryType(no)-couchbase": Object {
    "applicationWithEntities": Object {
      "config": Object {
        "applicationType": "monolith",
        "authenticationType": "jwt",
        "baseName": "jhipster",
        "buildTool": "gradle",
        "databaseType": "couchbase",
        "dtoSuffix": "Rest",
        "enableTranslation": true,
        "entitySuffix": "",
        "jhiPrefix": "fix",
        "languages": Array [
          "fr",
          "en",
        ],
        "nativeLanguage": "en",
        "packageName": "com.package",
        "reactive": true,
        "searchEngine": "couchbase",
        "serverSideOptions": Array [],
        "serviceDiscoveryType": "no",
        "skipCommitHook": true,
        "skipUserManagement": true,
        "testFrameworks": Array [
          "cucumber",
        ],
      },
      "entities": Array [
        Object {
          "changelogDate": "20220129000100",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "Simple",
        },
        Object {
          "changelogDate": "20220129000200",
          "clientRootFolder": "test-root",
          "dto": "mapstruct",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "AnotherSimple",
          "pagination": "pagination",
          "service": "serviceImpl",
        },
      ],
    },
    "defaults": true,
  },
  "monolith-oauth2-reactive(false)-maven-enableTranslation(false)-tech.jhipster-jhi-Entity-DTO-skipCommitHook(false)-websocket(false)-serviceDiscoveryType(no)-searchEngine(no)": Object {
    "applicationWithEntities": Object {
      "config": Object {
        "applicationType": "monolith",
        "authenticationType": "oauth2",
        "baseName": "jhipster",
        "buildTool": "maven",
        "databaseType": "couchbase",
        "dtoSuffix": "DTO",
        "enableTranslation": false,
        "entitySuffix": "Entity",
        "jhiPrefix": "jhi",
        "languages": Array [
          "fr",
          "en",
        ],
        "nativeLanguage": "en",
        "packageName": "tech.jhipster",
        "reactive": false,
        "searchEngine": "no",
        "serverSideOptions": Array [],
        "serviceDiscoveryType": "no",
        "skipCommitHook": false,
        "testFrameworks": Array [],
        "websocket": false,
      },
      "entities": Array [
        Object {
          "changelogDate": "20220129000100",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "Simple",
        },
        Object {
          "changelogDate": "20220129000200",
          "clientRootFolder": "test-root",
          "dto": "mapstruct",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "AnotherSimple",
          "pagination": "pagination",
          "service": "serviceImpl",
        },
      ],
    },
    "defaults": true,
  },
  "monolith-oauth2-reactive(true)-gradle-enableTranslation(true)-com.package-fix--Rest-skipCommitHook(true)-cucumber-enableSwaggerCodegen:true-eureka-couchbase": Object {
    "applicationWithEntities": Object {
      "config": Object {
        "applicationType": "monolith",
        "authenticationType": "oauth2",
        "baseName": "jhipster",
        "buildTool": "gradle",
        "databaseType": "couchbase",
        "dtoSuffix": "Rest",
        "enableTranslation": true,
        "entitySuffix": "",
        "jhiPrefix": "fix",
        "languages": Array [
          "fr",
          "en",
        ],
        "nativeLanguage": "en",
        "packageName": "com.package",
        "reactive": true,
        "searchEngine": "couchbase",
        "serverSideOptions": Array [
          "enableSwaggerCodegen:true",
        ],
        "serviceDiscoveryType": "eureka",
        "skipCommitHook": true,
        "testFrameworks": Array [
          "cucumber",
        ],
      },
      "entities": Array [
        Object {
          "changelogDate": "20220129000100",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "Simple",
        },
        Object {
          "changelogDate": "20220129000200",
          "clientRootFolder": "test-root",
          "dto": "mapstruct",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "AnotherSimple",
          "pagination": "pagination",
          "service": "serviceImpl",
        },
      ],
    },
    "defaults": true,
  },
  "monolith-session-reactive(false)-maven-enableTranslation(false)-tech.jhipster-jhi-Entity-DTO-skipCommitHook(false)-gatling-websocket(false)-skipUserManagement(false)-serviceDiscoveryType(no)-searchEngine(no)": Object {
    "applicationWithEntities": Object {
      "config": Object {
        "applicationType": "monolith",
        "authenticationType": "session",
        "baseName": "jhipster",
        "buildTool": "maven",
        "databaseType": "couchbase",
        "dtoSuffix": "DTO",
        "enableTranslation": false,
        "entitySuffix": "Entity",
        "jhiPrefix": "jhi",
        "languages": Array [
          "fr",
          "en",
        ],
        "nativeLanguage": "en",
        "packageName": "tech.jhipster",
        "reactive": false,
        "searchEngine": "no",
        "serverSideOptions": Array [],
        "serviceDiscoveryType": "no",
        "skipCommitHook": false,
        "skipUserManagement": false,
        "testFrameworks": Array [
          "gatling",
        ],
        "websocket": false,
      },
      "entities": Array [
        Object {
          "changelogDate": "20220129000100",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "Simple",
        },
        Object {
          "changelogDate": "20220129000200",
          "clientRootFolder": "test-root",
          "dto": "mapstruct",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "AnotherSimple",
          "pagination": "pagination",
          "service": "serviceImpl",
        },
      ],
    },
    "defaults": true,
  },
  "monolith-session-reactive(true)-gradle-enableTranslation(true)-com.package-fix--Rest-skipCommitHook(true)-gatling-enableSwaggerCodegen:true-skipUserManagement(true)-consul-couchbase": Object {
    "applicationWithEntities": Object {
      "config": Object {
        "applicationType": "monolith",
        "authenticationType": "session",
        "baseName": "jhipster",
        "buildTool": "gradle",
        "databaseType": "couchbase",
        "dtoSuffix": "Rest",
        "enableTranslation": true,
        "entitySuffix": "",
        "jhiPrefix": "fix",
        "languages": Array [
          "fr",
          "en",
        ],
        "nativeLanguage": "en",
        "packageName": "com.package",
        "reactive": true,
        "searchEngine": "couchbase",
        "serverSideOptions": Array [
          "enableSwaggerCodegen:true",
        ],
        "serviceDiscoveryType": "consul",
        "skipCommitHook": true,
        "skipUserManagement": true,
        "testFrameworks": Array [
          "gatling",
        ],
      },
      "entities": Array [
        Object {
          "changelogDate": "20220129000100",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "Simple",
        },
        Object {
          "changelogDate": "20220129000200",
          "clientRootFolder": "test-root",
          "dto": "mapstruct",
          "fields": Array [
            Object {
              "fieldName": "simpleName",
              "fieldType": "String",
            },
          ],
          "name": "AnotherSimple",
          "pagination": "pagination",
          "service": "serviceImpl",
        },
      ],
    },
    "defaults": true,
  },
}
`;
